python/ql/src/semmle/python/dataflow/new/internal/DataFlowPrivate.qll

```diff
@@ -1322,7 +1360,7 @@ module IterableUnpacking {
    */
   predicate iterableUnpackingForReadStep(CfgNode nodeFrom, Content c, Node nodeTo) {
     exists(ForTarget target |
-      nodeFrom.asExpr() = target.getSource() and
+      nodeFrom.getNode().getNode() = target.getSource() and
```

Comment:
I'm super confused about why you had to make this change?

I guess that the other iterable unpacking / for handling predicates had to be changed since `Node` was to relaxed, since that could allow a `RecursiveElement` and you didn't want that... but this one, I simply can't figure out :smile:

Reply:
As I recall, it gives non-monotonic recursion. I think it is because `asExpr` is defined in terms of a cast which then checks class membership.

Reply:
Alright. :shrug: